libraries/AP_RangeFinder/AP_RangeFinder_analog.cpp

```diff
@@ -106,11 +106,13 @@ void AP_RangeFinder_analog::update(void)
         } else {
             dist_m = scaling / (v - offset);
         }
-        if (dist_m > _max_distance_cm * 0.01f) {
-            dist_m = _max_distance_cm * 0.01f;
-        }
         break;
     }
+    if (dist_m > _max_distance_cm * 0.01f) {
```

Comment:
I think just removing the constrain is correct, there should be no need to also add this check.

This is the same as in `update_status`:

https://github.com/ArduPilot/ardupilot/blob/be80cb12effa144f411f926d39c18890c7bd9ad1/libraries/AP_RangeFinder/AP_RangeFinder_Backend.cpp#L59

So I don't think would it trigger here and not trigger there.

Reply:
it creates the trigger and keeps the return from being a huge number...similar pattern in other backends....

Reply:
Hum, in that case we could move the constrain to the `update_status` function so it works for all backends. 

As you say somewhere might not be checking the status but we really should fix that.

Reply:
that would be a massive re-write of a lot of code! most consumers never check if the distance is valid!  beyond the scope of this patch....constraining here would be a safeguard for future drivers...will discuss during review call...

Reply:
I think that most consumers checks the distance is valid.  Do you have an example of one that doesn't?

Reply:
I agree with @IamPete1 that we shouldn't be limiting the distance here because it's an unnecessary loss of information.  There's nothing wrong with recording/reporting a distance longer than distance_max() as long as the status is changed to out-of-range-high.

The case where we set the distance to distance_max()+1 is when the driver tells us in another way that the reading is invalid.  So for example some lidars return 65535 (or some other magic number) that is not a distance at all but instead essentially an error code.

The only downside is that we can end up with the strange case where the user see (or the logs record) the range reported from the lidar getting shorter as it goes out of range.  So for example imagine a lidar that has a reliable range of 10m (so the user sets RNGFND1_MAX_CM = 11m) and an absolute range of 20m and above this the lidar itself returns 65535.  In this case, if the user has the rangefinder range displayed on their ground station they may see the reported range climb above 10m (which is OK, perhaps the numbers get noisy) but then once it gets above 20m it suddenly jumps back down to 11m.  This is a bit odd.. but that's how it is at the moment I think.

I also agree with @IamPete1 that if we don't like the odd behaviour above we should change it in the AP_RangeFinder_Backend::update_status() method.